v-next/hardhat/src/types/solidity/build-system.ts

```diff
@@ -147,13 +147,9 @@ export type FileBuildResult =
 
 export interface CacheHitFileBuildResult {
   type: FileBuildResultType.CACHE_HIT;
-  // TODO: Should we remove this? It is a buildId of an already existing build
-  // info.
-  // NOTE: The buildId and contractArtifactsGenerated are useful when one uses
-  // the build system programatically and wants to find out what artifacts
-  // were generated for a given file, and with what configuration.
   buildId: string;
   contractArtifactsGenerated: string[];
+  warnings: CompilerOutputError[];
```

Comment:
Since we're storing the entire compiler outputs in the cache, we're able to extra warnings out of them even when they're retrieved from cache.

Reply:
Nice! 

Note to self: are we remapping these?